### DISCUSSION TITLE
Chessboard scanner

### DISCUSSION THREAD

**Discussion by arlegotin** [lichess-org/mobile#716] [Ideas]
Hi everyone,

I'm developing [a package](https://github.com/arlegotin/chessml) for detecting and recognizing chessboards from images or videos. It retrieves FEN, which can further be used for analysis. Would you be interested in integrating this into your app?

https://github.com/lichess-org/mobile/assets/1470560/a26bae37-2d21-4538-bb64-a9bf2c7a429c

**Comment by veloce** [comment]
Hi! 
This is interesting but we believe it would be better if it stays as a separate app. It could integrate well with the lichess app by charging a FEN. And we don't have currently the resources to support the maintenance cost for such a feature.

**Reply by arlegotin** [reply]
Of course, the model returns FEN as its base, which can further be used for analysis; the rendered board image is just for demonstration purposes.

**Comment by ijm8710** [comment]
Oh man @veloce would absolutely love something like this. Any reason why something like a chess clock is cool to have as a tool but not a board reader?

**Comment by princip** [comment]
So happy @veloce is building this! I had the same idea recently don't really have the skillset to build it!
Do you plan to develop it to include real chess boards as well as pictures of chess boards

**Reply by arlegotin** [reply]
@princip recognizing real chessboards is a different challenge, so I'm not working on it now, but maybe in the future!

**Reply by ijm8710** [reply]
> @princip recognizing real chessboards is a different challenge, so I'm not working on it now, but maybe in the future!

Hey are you still working on this/do you plan to release an independent app?

Super sad they turned you down this far to incorporate with lichess but I'd be interested none the less

**Reply by arlegotin** [reply]
@ijm8710 I’m not sure about the whole app right now, but I could turn it into an API if you're interested

**Reply by ijm8710** [reply]
> @ijm8710 I’m not sure about the whole app right now, but I could turn it into an API if you're interested

Appreciate the offer. I asked chat gpt if it would be relatively easy to consume an api without a coding background. They said it could be challenging but one of ideas they suggested was that I could potentially use it thru Apple shortcuts (depending on the api).
Do you think that's possible or unlikely. Def interested if there wasn't a huge jump to understand how to use it

**Reply by arlegotin** [reply]
@ijm8710 I'm sure there are already apps that do the scanning, like Chessify, Chessvision, or others

**Reply by ijm8710** [reply]
> @ijm8710 I'm sure there are already apps that do the scanning, like Chessify, Chessvision, or others

Thanks yeah I used to use chessify but they went heavily premium and then even chessvision has a good bit of premium. If there were ads/vids to watch more or a brief cool down period, chesvision would still be my go-to but the cooldown is for an entire day which is not amazing

This is why I would've loved yours in liches sticking to their open source model
